runhouse/servers/cluster_servlet.py

```diff
@@ -217,3 +234,82 @@ async def aclear_all_references_to_env_servlet_name(self, env_servlet_name: str)
+                    "data": status,
+                }
+                cluster_uri = (
+                    status.get("cluster_config").get("name")[1:].replace("/", ":")
+                )
+                api_server_url = status.get("cluster_config").get(
+                    "api_server_url", rns_client.api_server_url
+                )
+                post_status_data_resp = requests.post(
+                    f"{api_server_url}/resource/{cluster_uri}/cluster/status",
+                    data=json.dumps(status_data),
+                    headers=rns_client.request_headers(),
+                )
+                if post_status_data_resp.status_code != 200:
+                    logger.error(
+                        f"({post_status_data_resp.status_code}) Failed to send cluster status to Den: {post_status_data_resp.text}"
+                    )
+                logger.info(
+                    f"Successfully updated cluster status in Den. Next ping to den in {round(DEFAULT_POST_STATUS_TO_DEN_INTERVAL_SIZE / 60, 2)} minutes."
+                )
+                interval_size = status.get("cluster_config").get(
+                    "den_status_ping_interval"
+                )
+            except Exception as e:
+                logger.error(
+                    f"Failed to post cluster status to den: {e}. Please check cluster logs for more info."
+                )
+                logger.info(
+                    "Temporarily increasing the interval between two status pings to den. Next ping will be in an hour."
+                )
+                await asyncio.sleep(3600)
+            finally:
+
+                await asyncio.sleep(interval_size)
+
+    def send_status_info_to_den_sync(self):
+        asyncio.run(self.asend_status_info_to_den())
+
+    def schedule_post_status(self):
+        # adding post status to den thread
+        logger.info("adding send_status_info_to_den_sync to thread pool")
+        # delay the start of post_status_thread, so we'll finish the cluster startup properly
+        post_status_thread = threading.Timer(
+            DELAY_STATUS_POST_THREAD, self.send_status_info_to_den_sync
+        )
+        logger.info("starting send_status_info_to_den_sync thread")
+
+        post_status_thread.start()
+
+    def set_obj_store(self, associated_obj_store):
+        self.obj_store = associated_obj_store
+
+    async def astatus(self):
+        if not self.obj_store:
+            logger.error(
+                "The relevant obj-store is not associated with the cluster servlet, therefore can't get cluster status."
+                " Setting the obj_store can be done using cluster_servlet.set_obj_store(associated_obj_store)."
+            )
+            raise ClusterServletError(
+                "Can't find the obj_store associated with current ClusterServlet."
+            )
+        return await self.obj_store.astatus()
+
+    def is_config_yaml_exists(self):
```

Comment:
do we need a helper here (makes more sense a general util i think but also is this re-used in more than one place?)

Reply:
let's just throw this into the top level `utils.py`